homeassistant/components/lcn/binary_sensor.py

```diff
@@ -83,6 +84,23 @@ def __init__(self, config: ConfigType, config_entry: ConfigEntry) -> None:
     async def async_added_to_hass(self) -> None:
         """Run when entity about to be added to hass."""
         await super().async_added_to_hass()
+
+        async_create_issue(
+            self.hass,
+            DOMAIN,
+            "deprecated_regulatorlock_sensor",
+            breaks_in_ha_version="2025.2.0",
+            is_fixable=False,
+            is_persistent=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_regulatorlock_sensor",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "LCN",
+            },
+        )
```

Comment:
We should rather check if they are used in automations and scripts and raise if they are

Reply:
Could you please give me a hint how to do that?

Would you also briefly explain to me, what's wrong about raising the issue at the moment when the entities are set up? Raising the issue should not only occur if the entities are used in automations and scripts but generally if they are used (even from the frontend)?

Reply:
We usually get more false positives for users and we want issues to be actionable, not a "hey, did you know that ...". 

In the past we deprecated the harmony switch entities, I think that's quite a clean example. (It's already removed from the codebase, but the PRs are still there)

I can link when I finally booted my pc

Reply:
Thanks! I guess I found it here: https://github.com/home-assistant/core/pull/119206.
I'll implement it in the same way.